types/receipt.go

```diff
@@ -32,6 +32,16 @@ type Receipt struct {
 	GasUsed         uint64
 	ContractAddress *Address
 	TxHash          Hash
+
+	TransactionType TxType
```

Comment:
Do you need this field? As I can see it is only used for marshaling (`IsLegacyTx` is used there). On the other hand, `IsStateTx` isn't used anywhere.

Reply:
We gonna keep field for now but I have removed `IsLegacyTx` func